### PR TITLE
Add parallel_block capability to Phylanx

### PR DESCRIPTION
**Pull request by stevenrbrandt**
This PR allows one to write code containing parallel blocks. To use it, first import "parallel" from phylanx, then use "with parallel" or "with phylanx.parallel" as appropriate to generate a parallel_block() in physl. See test "parallel.py".

### CODE REVIEW

**Comment by hkaiser** [comment]
@stevenrbrandt please fix the flake errors reported.

File: python/phylanx/__init__.py [hunk 1]
```diff
@@ -6,3 +6,10 @@
 from .core import *
 from .core.config import *
 from .ast.transducer import Phylanx
+
+
+class parallel:
```
Comment (rtohid) [inline python/phylanx/__init__.py]: Please move this to an appropriate module- perhaps under execution_tree subpackage.

File: python/phylanx/ast/physl.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import inspect
 import phylanx.execution_tree
 from phylanx import compiler_state
+from phylanx.ast.utils import dump_info
```
Comment (rtohid) [inline python/phylanx/ast/physl.py]: This is not used, please remove it.

File: python/phylanx/ast/physl.py [hunk 3]
```diff
@@ -772,6 +773,20 @@ def _NestedSubscript(node):
         #         return [op, (value, slice_)]
         return [op, (value, slice_)]
 
+    def _With(self, node):
+        if 0 < len(node.items) and type(node.items[0]) == ast.withitem:
```
Comment (rtohid) [inline python/phylanx/ast/physl.py]: This should be handled in the `_Withitem` method.

Comment (stevenrbrandt) [reply python/phylanx/ast/physl.py]: Not sure how that can be done. The information needed for the block is not completely contained under ast.withitem.

File: python/phylanx/__init__.py (hunk 1 above)
Comment (stevenrbrandt) [reply python/phylanx/__init__.py]: Do you have a suggestion?

Comment (stevenrbrandt) [reply python/phylanx/__init__.py]: It needs to be in package phylanx.

**Pull request merged**